src/components/ContactPage/Form/Form.tsx

```diff
@@ -52,11 +62,53 @@ export class Form extends React.Component<undefined, undefined> {
                     </div>
                 </div>
                 <SubmitButton className="btn btn_primary"/>
+                <ReCAPTCHA
+                    ref={this.setRecaptchaElement}
+                    sitekey={Config.reCaptchaApiKey}
+                    onChange={this.handleSubmit}
+                    size="invisible"
+                    className="recaptcha-badge"
+                />
             </form>
         );
     }
 
-    private handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    // TODO: confirmation
+    // tslint:disable-next-line
+    protected handleSubmit = async () => {};
+
+    protected validateRecaptcha = (e: React.FormEvent<HTMLFormElement>) => {
         e.preventDefault();
+
+        this.recaptcha.execute();
+        this.setRecaptchaClasses();
     };
+
+    protected setRecaptchaElement = (el: ReCAPTCHA) => this.recaptcha = el;
+
+    protected setRecaptchaClasses() {
+        // frame
+        let element = document.body.querySelector("iframe[title~='recaptcha']");
+
+        if (!element) {
+            return;
+        }
+
+        document.body.classList.add("recaptcha-check");
+
+        (element.parentNode as HTMLElement).className = "iframe-wrap";
+
+        while ((element.parentNode as HTMLElement).tagName !== "BODY") {
```

Comment:
This code may cause infinite loop, because in some cases `a.parentNode === a` (when *a* node not appended anywhere).
Why not to use document.body?